kie-server-parent/kie-server-controller/kie-server-controller-impl/src/test/java/org/kie/server/controller/impl/storage/ServerTemplateStorageTest.java

```diff
@@ -0,0 +1,371 @@
+        assertEquals(this.processConfig.getKSession(), processConfig.getKSession());
+        assertEquals(this.processConfig.getMergeMode(), processConfig.getMergeMode());
+        assertEquals(this.processConfig.getRuntimeStrategy(), processConfig.getRuntimeStrategy());
+
+        // second container spec
+        spec = iterator.next();
+        assertNotNull(spec);
+
+        assertEquals(newContainerSpec.getId(), spec.getId());
+        assertEquals(newContainerSpec.getReleasedId(), spec.getReleasedId());
+        assertEquals(newContainerSpec.getServerTemplateKey().getId(), spec.getServerTemplateKey().getId());
+        if (isRequired()) {
+            assertEquals(newContainerSpec.getServerTemplateKey().getName(), spec.getServerTemplateKey().getName());
+        }
+        assertEquals(newContainerSpec.getConfigs().size(), spec.getConfigs().size());
+    }
+
+    protected void createServerTemplateWithContainer() {
+        serverTemplate = new ServerTemplate();
+
+        serverTemplate.setName("test server");
+        serverTemplate.setId(UUID.randomUUID().toString());
+
+        Map<Capability, ContainerConfig> configs = new HashMap<Capability, ContainerConfig>();
+        ruleConfig = new RuleConfig();
+        ruleConfig.setPollInterval(1000l);
+        ruleConfig.setScannerStatus(KieScannerStatus.STARTED);
+
+        configs.put(Capability.RULE, ruleConfig);
+
+        processConfig = new ProcessConfig();
+        processConfig.setKBase("defaultKieBase");
+        processConfig.setKSession("defaultKieSession");
+        processConfig.setMergeMode("MERGE_COLLECTION");
+        processConfig.setRuntimeStrategy("PER_PROCESS_INSTANCE");
+
+        configs.put(Capability.PROCESS, processConfig);
+
+        containerSpec = new ContainerSpec();
+        containerSpec.setId("test container");
+        containerSpec.setServerTemplateKey(new ServerTemplateKey(serverTemplate.getId(), serverTemplate.getName()));
+        containerSpec.setReleasedId(new ReleaseId("org.kie", "kie-server-kjar", "1.0"));
+        containerSpec.setStatus(KieContainerStatus.STOPPED);
+        containerSpec.setConfigs(configs);
+
+        serverTemplate.addContainerSpec(containerSpec);
+
+        container = new Container();
+        container.setServerInstanceId(serverTemplate.getId());
+        container.setServerTemplateId(serverTemplate.getId());
+        container.setResolvedReleasedId(containerSpec.getReleasedId());
+        container.setContainerName(containerSpec.getContainerName());
+        container.setContainerSpecId(containerSpec.getId());
+        container.setUrl("http://fake.server.net/kie-server");
+
+    }
+
+    protected void checkTemplateInstances(ServerTemplate fromStorage) {
+        Collection<ServerInstanceKey> instances = fromStorage.getServerInstanceKeys();
+        assertNotNull(instances);
+        assertEquals(0, instances.size());
+    }
+
+    protected boolean isRequired() {
```

Comment:
What does this condition mean? Please adjust the name to be more descriptive.

Reply:
Renamed the method and added short description. 

https://github.com/rhtevan/droolsjbpm-integration/blob/fd4635e8e9b2263458db3339d7ea0c94b7ae87b6/kie-server-parent/kie-server-controller/kie-server-controller-impl/src/test/java/org/kie/server/controller/impl/storage/ServerTemplateStorageTest.java#L368-L374